### PR TITLE
Fixes to NIP-46 based on my possibly incorrect understanding

### PR DESCRIPTION
**Pull request by mikedilger**
Implementing NIP-46, I ran into a number of things that were not clearly specified.  This is an attempt to clarify these.

In case I got these things wrong, please correct me.

@pablof7z

### CODE REVIEW

File: 46.md [hunk 1]
```diff
@@ -30,13 +30,13 @@ The client generates a QR code in the following form (URL-encoded):
 nostrconnect://<client-key-hex>?relay=wss://...&metadata={"name":"...", "url": "...", "description": "..."}
 ```
 
-The signer scans the QR code and sends a `connect` message to the client in the specified relays.
+The signer scans the QR code and sends a `connect` message to the client in the specified relays using the request event format.
```
Comment (fiatjaf) [inline 46.md:33]: what is the request event format?

Comment (fiatjaf) [reply 46.md:33]: oh, I see

maybe we should rephrase this to say "RPC call" somehow.

Comment (mikedilger) [reply 46.md:33]: Basically I'm just confused at how the signer can send a "connect" message to the client, when the signer's event format doesn't allow specifying a method.  It's a backwards command.  So I'm assuming it must use the other event format for this.  Given I don't actually know how this is supposed to be resolved I'm just not implementing this part of the spec for now.

Comment (mikedilger) [reply 46.md:33]: Ok I'm implementing 2. I commented out my impl of flow 1 since I couldn't understand how to send a connect from the server to the client.  I was hoping someone did that and could explain what has actually happened in practice.

File: 46.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ The signer key will always be the key of the user who controls the signer device
 ### Methods
 
 - **connect**
-  - params: [`pubkey`, `secret`]
+  - params: [`pubkey-hex`, `secret`]
```
Comment (monlovesmango) [inline 46.md:70]: given that connect can be from either the signer or the client, do we know whether `pubkey-hex` changes in each scenario?

also any idea why we need `secrect`? assuming this is the client's private key..?

Comment (pablof7z) [reply 46.md:70]: no, `secret` is a secret that can be provided out of band to the client from the signer; this is an easy way to identify the client in front of the signer.

Scenario:

A company provides authorization to 100 employees to sign using the conpany's key
They create 100 secrets and give a secret to each employee.
Each secret is redeemed without prior knowledge of which client-pubkey will do the redemption.

Comment (monlovesmango) [reply 46.md:70]: where is this detailed?

is the secret for `connect` that is from client to signer only? or is it also for `connect` that is from signer to client as well?

I don't think i'm understanding, from the example you shared it seems like it is the employee's private key that can sign on behalf of the company's key. which sounds like the private key on the client side.
